### PR TITLE
replace dependency DiffEq by StochasticDiffEq

### PR DESCRIPTION
**Pull request by haakon-e**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -45,7 +45,6 @@ ClimaCore = "0.2, 0.3, 0.4, 0.5"
 CloudMicrophysics = "0.3"
 Debugger = "0.6"
 Dierckx = "0.5"
-DifferentialEquations = "6.19"
```
Comment (charleskawczynski) [inline Project.toml:48]: Can we add a compat entry for `StochasticDiffEq`?

File: src/stochastic_closures.jl [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import Distributions
-import DifferentialEquations: SOSRI
+import StochasticDiffEq: SDEProblem, SOSRI, solve
```
Comment (charleskawczynski) [inline src/stochastic_closures.jl:2]:
```suggestion
import StochasticDiffEq
const SDEQ = StochasticDiffEq
```
Can we move this to `src/TurbulenceConvection.jl` and then just use `SDEQ.SDEProblem`/ `SDEQ.SOSRI` / `SDEQ.solve`?

**Review by charleskawczynski — COMMENTED** [review]
Left some suggested changes

**Comment by charleskawczynski** [comment]
Wow, here's a list of the deps we drop from this PR:

```
  [dce04be8] - ArgCheck v2.1.0
  [15f4f7f2] - AutoHashEquals v0.2.0
  [aae01518] - BandedMatrices v0.16.11
  [198e06fe] - BangBang v0.3.32
  [9718e550] - Baselet v0.1.1
  [e2ed5e7c] - Bijections v0.1.3
  [764a87c0] - BoundaryValueDiffEq v2.7.1
  [00ebfdb7] - CSTParser v3.3.0
  [861a8166] - Combinatorics v1.0.2
  [a80b9123] - CommonMark v0.8.5
  [b152e2b5] - CompositeTypes v0.1.2
  [a33af91c] - CompositionsBase v0.1.1
  [244e2a9f] - DefineSingletons v0.1.1
  [bcd4f6db] - DelayDiffEq v5.31.2
  [459566f4] - DiffEqCallbacks v2.18.0
  [5a0ffddc] - DiffEqFinancial v2.4.0
  [055956cb] - DiffEqPhysics v3.9.0
  [0c46a032] - DifferentialEquations v6.20.0
  [c619ae07] - DimensionalPlotRecipes v1.2.0
  [5b8099bc] - DomainSets v0.5.9
  [7c1d4256] - DynamicPolynomials v0.4.1
  [22cec73e] - InitialValues v0.2.11
  [98e50ef6] - JuliaFormatter v0.19.2
  [b964fa9f] - LaTeXStrings v1.3.0
  [23fbe1c1] - Latexify v0.15.9
  [e9d8d322] - Metatheory v1.3.2
  [128add7d] - MicroCollections v0.1.1
  [961ee093] - ModelingToolkit v7.1.3
  [f9640e96] - MultiScaleArrays v1.8.1
  [102ac46a] - MultivariatePolynomials v0.4.2
  [d8a4904e] - MutableArithmetics v0.3.1
  [f4b8ab15] - NonlinearSolvers v0.1.1
  [65888b18] - ParameterizedFunctions v5.12.2
  [42d2dcc6] - Referenceables v0.1.2
  [7e49a35a] - RuntimeGeneratedFunctions v0.5.3
  [1bc83da4] - SafeTestsets v0.0.1
  [171d559e] - SplittablesBase v0.1.14
  [9672c7b4] - SteadyStateDiffEq v1.6.6
  [c3572dad] - Sundials v4.6.0
  [d1185830] - SymbolicUtils v0.18.2
  [0c5d862f] - Symbolics v4.2.0
  [8ea1fca8] - TermInterface v0.2.3
  [ac1d9e8a] - ThreadsX v0.1.8
  [0796e94c] - Tokenize v0.5.21
  [28d57a85] - Transducers v0.4.67
  [5c2747f8] - URIs v1.3.0
  [1986cc42] - Unitful v1.9.2
  [fb77eaff] - Sundials_jll v5.2.1+0
  [bea87d4a] - SuiteSparse_jll
```

**Comment by charleskawczynski** [comment]
I'm going to push this forward.

File: src/closures/entr_detr.jl [hunk 3]
```diff
@@ -129,9 +128,9 @@ function entr_detr(param_set, εδ_model_vars, εδ_model_type::NNEntr)
 
     # Neural network closure
     nn_arc = (4, 2, 2)  # (#inputs, #neurons, #outputs)
-    nn_model = Chain(
-        Dense(reshape(c_gen[1:8], nn_arc[2], nn_arc[1]), c_gen[9:10], sigmoid),
-        Dense(reshape(c_gen[11:14], nn_arc[3], nn_arc[2]), c_gen[15:16], softplus),
+    nn_model = Flux.Chain(
```
Comment (yairchn) [inline src/closures/entr_detr.jl:131]: would this require a change in the guidelines in 
https://github.com/CliMA/TurbulenceConvection.jl/pull/653

@costachris ?

**Review by yairchn — APPROVED** [review]
left one small comment, otherwise LGTM

File: src/closures/entr_detr.jl (hunk 3 above)
Comment (charleskawczynski) [reply src/closures/entr_detr.jl:131]: The only change here is to qualify things. I left a comment [here](https://github.com/CliMA/TurbulenceConvection.jl/pull/653/files#r772734789)

**Comment by haakon-e** [comment]
thanks for taking the lead and merging this @charleskawczynski!

**Pull request merged**